connector/web3/web3.go

```diff
@@ -2,27 +2,39 @@
 package web3
 
 import (
+	"bytes"
+	"errors"
 	"fmt"
 	"github.com/dexidp/dex/connector"
 	"github.com/dexidp/dex/pkg/log"
 	"github.com/ethereum/go-ethereum/accounts"
+	"github.com/ethereum/go-ethereum/accounts/abi/bind"
 	"github.com/ethereum/go-ethereum/common"
 	"github.com/ethereum/go-ethereum/common/hexutil"
 	"github.com/ethereum/go-ethereum/crypto"
+	"github.com/ethereum/go-ethereum/ethclient"
+	"os"
 )
 
 type Config struct {
 	InfuraID string `json:"infuraId"`
-	RPCURL   string `json:"rpcUrl"`
 }
 
 func (c *Config) Open(id string, logger log.Logger) (connector.Connector, error) {
-	return &web3Connector{infuraID: c.InfuraID, rpcURL: c.RPCURL}, nil
+	w := &web3Connector{infuraID: c.InfuraID, logger: logger}
+
+	ethClient, err := createEthClient()
```

Comment:
small thing but it looks like we're only using `createEthClient` once, might make sense to bring that logic directly into this function rather than separating it out

Reply:
I decided to separate them to make sure the functions has a single responsibility and to make it easier to manage.

Reply:
You want to construct this from the config file, not an environment variable.